### PR TITLE
[tools] Make release script a bit more flexible

### PR DESCRIPTION
**Pull request by rmetzger**

### CODE REVIEW

**Comment by uce** [comment]
Good changes +1

There are some assumptions about the call order of the newly introduced functions though (like you have to call prepare make_src_release [or be in the checked out repo] in that order). I guess it's fine, we don't want to over engineer this ;)

**Comment by hsaputra** [comment]
HI @rmetzger, could you summarize the intention of the PR here? Like what is the final goal of the changes?

**Comment by mxm** [comment]
@hsaputra This is just refactoring. I think your changes improve the readability of the code. Like Ufuk mentioned, it does not necessarily make the script more flexible. Due to side effects of the function calls they need to be called in a specific order. It does make the script "a bit more flexible" though :)

File: tools/create_release_files.sh [hunk 1]
```diff
@@ -138,7 +146,7 @@ make_binary_release() {
     echo "Files in uberjar: $NUM_FILES_IN_UBERJAR. Uberjar: $UBERJAR"
     if [ "$NUM_FILES_IN_UBERJAR" -ge "65536" ] ; then
       echo "The number of files in the uberjar ($NUM_FILES_IN_UBERJAR) exceeds the maximum number of possible files for Java 6 (65536)"
-      exit 1
+    #  exit 1
```
Comment (mxm) [inline tools/create_release_files.sh]: Shouldn't we fail here? What's the point of the check after all?

File: tools/create_release_files.sh [hunk 2]
```diff
@@ -85,37 +87,43 @@ else
...
-else
-    find . -name 'pom.xml' -type f -exec sed -i    's#<version>'$OLD_VERSION'</version>#<version>'$NEW_VERSION'</version>#' {} \;
-fi
+make_source_release() {
 
-git commit --author="$GIT_AUTHOR" -am "Commit for release $RELEASE_VERSION"
-git remote add asf_push https://$USER_NAME@git-wip-us.apache.org/repos/asf/flink.git
-RELEASE_HASH=`git rev-parse HEAD`
-echo "Echo created release hash $RELEASE_HASH"
+  #find . -name 'pom.xml' -type f -exec sed -i 's#<version>$OLD_VERSION</version>#<version>$NEW_VERSION</version>#' {} \;
```
Comment (mxm) [inline tools/create_release_files.sh]: I think you can remove this line.

**Comment by hsaputra** [comment]
Thanks @mxm , yeah I guess it is a refactoring. I was trying to figure out what changes make it more flexible.

**Comment by rmetzger** [comment]
I needed to make these changes to the script when I created the additional binaries for hadoop24,hadoop26,... 
For that I needed to create binaries without deploying stuff to maven again. Thats why I put these operations into separate functions so that I can control whether they are executed or not.

I'll merge this change now.
